src/System.Drawing.Common/src/System/Drawing/Font.Windows.cs

```diff
@@ -311,17 +311,22 @@ public unsafe static Font FromLogFont(object lf, IntPtr hdc)
             // Now that we know the marshalled size is the same as LOGFONT, copy in the data
             logFont = new SafeNativeMethods.LOGFONT();
 
-            if (!type.IsValueType)
-            {
-                // Only works with non value types
-                Marshal.StructureToPtr(lf, new IntPtr(&logFont), fDeleteOld: false);
-            }
-            else
+            try
             {
                 GCHandle handle = GCHandle.Alloc(lf, GCHandleType.Pinned);
                 Buffer.MemoryCopy((byte*)handle.AddrOfPinnedObject(), &logFont, nativeSize, nativeSize);
```

Comment:
` Marshal.StructureToPtr` has the exact same path for blittable structs. You can just call `Marshal.StructureToPtr` all the time now and skip this "optimization". It will be both simpler and likely faster too - because of `Marshal.StructureToPtr` does not allocate the GCHandle internally.

Reply:
It failed for me when I tried this before- did something change?

Reply:
Something must have. Perhaps I was hitting some runtime bug when I did this originally. 🤷🏻‍♂️

Reply:
I am not aware of any reason why that would not work.

I have tried https://github.com/dotnet/corefx/pull/42011/ locally and all tests are passing on it for me locally.

Reply:
> hitting some runtime bug

Doubt it. Nobody touched this stuff for a very long time.

Reply:
Clearly I'm hallucinating. :)